SecurityException at cgeo.Downloader

Google Play report for 2021.03.21-RC:

1 user, Android 11

```
java.lang.RuntimeException: 
  at android.app.ActivityThread.performLaunchActivity (ActivityThread.java:3792)
  at android.app.ActivityThread.handleLaunchActivity (ActivityThread.java:3968)
  at android.app.servertransaction.LaunchActivityItem.execute (LaunchActivityItem.java:85)
  at android.app.servertransaction.TransactionExecutor.executeCallbacks (TransactionExecutor.java:135)
  at android.app.servertransaction.TransactionExecutor.execute (TransactionExecutor.java:95)
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java:2307)
  at android.os.Handler.dispatchMessage (Handler.java:106)
  at android.os.Looper.loop (Looper.java:246)
  at android.app.ActivityThread.main (ActivityThread.java:8506)
  at java.lang.reflect.Method.invoke (Native Method)
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:602)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:1130)
Caused by: java.lang.SecurityException: 
  at android.os.Parcel.createExceptionOrNull (Parcel.java:2385)
  at android.os.Parcel.createException (Parcel.java:2369)
  at android.os.Parcel.readException (Parcel.java:2352)
  at android.database.DatabaseUtils.readExceptionFromParcel (DatabaseUtils.java:190)
  at android.database.DatabaseUtils.readExceptionWithFileNotFoundExceptionFromParcel (DatabaseUtils.java:153)
  at android.content.ContentProviderProxy.openTypedAssetFile (ContentProviderNative.java:781)
  at android.content.ContentResolver.openTypedAssetFileDescriptor (ContentResolver.java:1986)
  at android.content.ContentResolver.openAssetFileDescriptor (ContentResolver.java:1801)
  at android.content.ContentResolver.openInputStream (ContentResolver.java:1478)
  at cgeo.geocaching.downloader.ReceiveMapFileActivity.lambda$onCreate$0 (ReceiveMapFileActivity.java:71)
  at cgeo.geocaching.downloader.ReceiveMapFileActivity.lambda$onCreate$0$ReceiveMapFileActivity (Unknown Source)
  at cgeo.geocaching.downloader.-$$Lambda$ReceiveMapFileActivity$P-4IOTNyqWeNZKiZSuAPSN7CYwM.run (Unknown Source:4)
  at cgeo.geocaching.downloader.MapDownloaderUtils$1.execute (MapDownloaderUtils.java:122)
  at cgeo.geocaching.permission.PermissionHandler.requestStoragePermission (PermissionHandler.java:81)
  at cgeo.geocaching.downloader.MapDownloaderUtils.checkMapDirectory (MapDownloaderUtils.java:115)
  at cgeo.geocaching.downloader.ReceiveMapFileActivity.onCreate (ReceiveMapFileActivity.java:67)
  at android.app.Activity.performCreate (Activity.java:8198)
  at android.app.Activity.performCreate (Activity.java:8182)
  at android.app.Instrumentation.callActivityOnCreate (Instrumentation.java:1309)
  at android.app.ActivityThread.performLaunchActivity (ActivityThread.java:3765)
```

--- a/main/src/cgeo/geocaching/downloader/ReceiveMapFileActivity.java
+++ b/main/src/cgeo/geocaching/downloader/ReceiveMapFileActivity.java
@@ -83,7 +83,7 @@ public class ReceiveMapFileActivity extends AbstractActivity {
                             }
                         }
                     }
-                } catch (IOException e) {
+                } catch (IOException | SecurityException e) {
                     // ignore ZIP errors
                 }
                 // if no ZIP file: continue with copying the file
@@ -209,6 +209,9 @@ public class ReceiveMapFileActivity extends AbstractActivity {
                 } else {
                     status = doCopy(inputStream, outputUri);
                 }
+            } catch (SecurityException e) {
+                Log.e("SecurityException on receiving map file: " + e.getMessage());
+                return CopyStates.FILENOTFOUND_EXCEPTION;
             } catch (FileNotFoundException e) {
                 return CopyStates.FILENOTFOUND_EXCEPTION;
             } finally {
@@ -217,7 +220,6 @@ public class ReceiveMapFileActivity extends AbstractActivity {
 
             // clean up and refresh available map list
             if (!cancelled.get()) {
-                status = CopyStates.SUCCESS;
                 try {
                     getContentResolver().delete(uri, null, null);
                 } catch (IllegalArgumentException iae) {

